### ISSUE TITLE
self service create new user email notification flow

### ISSUE THREAD

**Issue by brandonopened** [opensalt/opensalt#343]
![email flow](https://user-images.githubusercontent.com/14943151/35687123-cdde316c-0765-11e8-88dd-37b1c250e92c.PNG)

**Comment by roverwolf** [comment]
Assumptions:
 - The "self service create new user" option would only be enabled and available if there is an email server configured.

Questions:
 - The image makes it look like the user can create an organisation, I am assuming this is not the case and must choose from the current list.
 - What happens if the Admin deletes the user instead of aproving or suspending the user?

**Comment by brandonopened** [comment]
correct, there will be an option to turn this on or off as on cfdoc there will be a button to 'create a new user'. 
https://opensalt.mybalsamiq.com/projects/opensalt/home
and
https://opensalt.mybalsamiq.com/projects/opensalt/Create%20an%20account-BA

- validation on the username not being a duplicate
- validation that the passwords are matching
- validation that the org name they enter is not a duplicate

Yes, we are saying a new org will be created with pending status as well
https://opensalt.mybalsamiq.com/projects/opensalt/Org%20Admin
(and same code for pending etc)

BUT the main 'action' is here on the user list page
https://opensalt.mybalsamiq.com/projects/opensalt/Org%20Admin

- When an admin approves a user their acccount becomes active. 
- If they had created a new org it will also be approved when the first user is approved

eventually we will send the notification to the org admin as selected on new user account creation, for now it will go to super user. (eg meta-admin).

**Comment by jmarks** [comment]
Additional requirement: Please make this feature a admin configurable application option. We have users who will not want auto registration and will require only SSO authentication of manged account creation. This shoudl be considered along with SSO integration stories.

**Comment by roverwolf** [comment]
#375 has been added as a first step to add the ability to do emails
